fds/services/dvc_service.py

```diff
@@ -187,28 +208,33 @@ def __add(self, add_argument: str):
                 # because he doesn't want the entire directory to be added.
 
                 # If the root is skipped because it is below threshold size then we don't need to check files
-                if root in skipped_dirs:
+                if root in ignored_dirs:
                     continue
                 # Then check files
                 for file in files:
-                    (file_to_add, skipped_dir) = self.__get_to_add_to_dvc(f"{root}/{file}", [], "File")
-                    if skipped_dir is not None:
-                        skipped_dirs.append(skipped_dir)
-                    if file_to_add is not None:
-                        chosen_files_or_folders.append(file_to_add)
+                    add_to_dvc = self.__get_to_add_to_dvc(f"{root}/{file}", [], "File")
+                    if add_to_dvc.file_to_ignore is not None:
+                        ignored_dirs.append(add_to_dvc.file_to_ignore)
+                    if add_to_dvc.file_to_add is not None:
+                        chosen_files_or_folders.append(add_to_dvc.file_to_add)
+                    if add_to_dvc.file_to_skip is not None:
+                        skipped_dirs.append(add_to_dvc.file_to_skip)
         self.logger.debug(f"Chosen folders to be added to dvc are {chosen_files_or_folders}")
-        if len(chosen_files_or_folders) == 0:
-            return "Nothing to add in DVC"
+        if len(chosen_files_or_folders) > 0:
+            self.printer.warn("Adding to dvc...")
+            progress_tracker = Bar('Processing', max=len(chosen_files_or_folders))
+            for add_to_dvc in chosen_files_or_folders:
+                execute_command(["dvc", "add", add_to_dvc])
+                progress_tracker.next()
```

Comment:
Wouldn't it be faster to do something like `dvc add "chosen_files_or_folders[0]" "chosen_files_or_folders[1]" ...`?

Reply:
Its easy to track this way, so I preferred doing it in the loop

Reply:
Easier to track but much much much slower